File: src/lib/scala/Slate.Common/src/main/scala/slate/common/envs/EnvSupport.scala
/**
  * <slate_header>
  * author: Kishore Reddy
  * url: https://github.com/kishorereddy/scala-slate
  * copyright: 2016 Kishore Reddy
  * license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  * desc: a scala micro-framework
  * usage: Please refer to license on github for more info.
  * </slate_header>
  */
package slate.common.envs


trait EnvSupport {


  def isProd:Boolean = isEnv(Prod)


  def isUat:Boolean = isEnv(Uat)


  def isQa:Boolean = isEnv(Qa)


  def isDev:Boolean = isEnv(Dev)


  def isDis:Boolean = isEnv(Dis)


  def isEnv(mode: String):Boolean = ???


  def isEnv(mode: EnvMode):Boolean = ???
}

File: src/lib/kotlin/slatekit-common/src/main/kotlin/slatekit/common/envs/EnvSupport.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */

package slatekit.common.envs

interface EnvSupport {


    val isProd: Boolean get() = isEnv(Prod)


    val isUat: Boolean get() = isEnv(Uat)


    val isQa: Boolean get() = isEnv(Qa)


    val isDev: Boolean get() = isEnv(Dev)


    val isDis: Boolean get() = isEnv(Dis)


    fun isEnv(envMode: EnvMode): Boolean


    fun isEnvName(envMode: String): Boolean
}
